### PR TITLE
RAP-4090

### PR DESCRIPTION
**Pull request by scarabus**

### CODE REVIEW

File: Libs/Reflex/src/main/java/reflex/function/DefinedNode.java [hunk 1]
```diff
@@ -37,21 +40,40 @@
  */
 public class DefinedNode extends BaseNode {
 
-    private String varName;
+    private String varName = null;
     private String namespacePrefix;
+    private ReflexNode node = null;
```
Comment (dukenguyen) [inline Libs/Reflex/src/main/java/reflex/function/DefinedNode.java]: this does not need to be set to null, same as 2 lines above

File: Libs/RaptureCore/src/test/java/rapture/kernel/scripting/ReflexParamsTest.java [hunk 2]
```diff
@@ -44,89 +46,139 @@
...
+    // public void testRAP4090() {
+    // // String scriptUri = makeScript("rap4090", "if defined(web['foo']) do \n" + " println(\"web['foo'] is defined \" + web['foo']); return \"def\";\n"
+    // // + "else do \n" + " println(\"web['foo'] is undefined\"); return \"notdef\";\n" + "end");
+    // String scriptUri = makeScript("rap4090", "a = defined(web['foo']);");
+    // String retval = Kernel.getScript().runScript(ctx, scriptUri, new HashMap<String, String>());
+    // assertEquals("notdef", retval);
+    // }
+
+    @Test
+    public void testRAP4090a() {
```
Comment (dukenguyen) [inline Libs/RaptureCore/src/test/java/rapture/kernel/scripting/ReflexParamsTest.java]: you should name your tests with what you are actually testing.  people looking at error logs dont want to go to jira

File: Libs/RaptureCore/src/test/java/rapture/kernel/scripting/ReflexParamsTest.java [hunk 3]
```diff
@@ -44,89 +46,139 @@
...
     @Test
     public void testParamsNotDefined() {
-        String retval = Kernel.getScript().runScript(ctx, uri + "/" + scr, new HashMap<String, String>());
+        String scriptUri = makeScript("testScript1", "if defined(_params) do \n" + "  println(\"params is defined \" + _params); return \"def\";\n"
+                + "else do \n" + "  println(\"params is undefined\"); return \"notdef\";\n" + "end");
+        String retval = Kernel.getScript().runScript(ctx, scriptUri, new HashMap<String, String>());
         assertEquals("notdef", retval);
     }
 
+    // @Test
```
Comment (dukenguyen) [inline Libs/RaptureCore/src/test/java/rapture/kernel/scripting/ReflexParamsTest.java]: delete comments

Comment (dukenguyen) [inline Libs/RaptureCore/src/test/java/rapture/kernel/scripting/ReflexParamsTest.java]: why is this a recurring theme?

Comment (scarabus) [inline Libs/RaptureCore/src/test/java/rapture/kernel/scripting/ReflexParamsTest.java]: Testing different code paths.

**Pull request merged**